pkg/plugin/sdk/v1alpha2/plugin/interface.go

```diff
@@ -62,3 +74,56 @@ func (*Base) PostTestHook(bool, ioutils.PromptRunner) error {
 func (*Base) PostRunHook(bool, ioutils.PromptRunner) error {
 	return nil
 }
+
+type CustomCommandFn (func(ctx context.Context, args []string, bzl bazel.Bazel) error)
+
+type Command struct {
+	*proto.Command
+	Run CustomCommandFn
+}
+
+func NewCommand(
+	use string,
+	shortDesc string,
+	longDesc string,
+	run func(ctx context.Context, args []string, bzl bazel.Bazel) error,
+) *Command {
+	return &Command{
+		Command: &proto.Command{
+			Use:       use,
+			ShortDesc: shortDesc,
+			LongDesc:  longDesc,
+		},
+		Run: run,
+	}
+}
+
+type CommandManager interface {
+	Save(commands []*Command) error
+	Execute(command string, ctx context.Context, args []string) error
+}
+
+type PluginCommandManager struct {
+	commands map[string]CustomCommandFn
+}
+
+func (cm *PluginCommandManager) Save(commands []*Command) error {
+	for _, cmd := range commands {
+		if _, exists := cm.commands[cmd.Use]; exists {
+			return fmt.Errorf("command '%s' is declared more than once by plugin", cmd.Use)
```

Comment:
The `%q` is very helpful in Go when you want to quote a string when formatting it.

```suggestion
			return fmt.Errorf("command %q is declared more than once by plugin", cmd.Use)
```